src/app/dashboard/under-notice/under-notice.component.html

```diff
@@ -1,4 +1,9 @@
-<div class="nhsuk-table-responsive" *ngIf="trainees$ | async as trainees">
+<ng-template #loading><mat-spinner #loading></mat-spinner></ng-template>
```

Comment:
when you say in app component, do you mean a component to wrap `mat-spinner`?

how would the http interceptor relate to the spinner? previously i've used http interceptors to append or reuse jwt token or user info on api calls... in this pr im showing the spinner whilst the `trainee$` observable hasn't resolved. or am i being daft and missing your point completely? 

`RouterModule` and `CommonModule` are currently imported within `SharedModule` and then the `DashboardModule` imports the `SharedModule`. what is the benefit of adding a spinner on route change?

Reply:
It can be in wrapper component that imports mat-spinner and angular cdk i.e. backdrop overlay.
A singleton service ideally injected in root will be used to subscribe to a behavior subject to hide show.
Inteceptor will intercept all httpClient accross the app (instead of just the one) and show hide while in progress.
router navigation observables can show on routing start (pipe) and hide on end / error / abort ..
The magic may occur on debouce, scan, etc.. where race conditions may occur on multiple http combined with routing etc. so one global overlay will block UI and spinner will show until last observable is resolved.

examples: https://medium.com/@velenra/using-angular-material-spinner-with-cdk-overlay-8ab92bfbafee

https://onthecode.co.uk/angular-display-spinner-every-request/

Reply:
ah i understand you now. but do we want to block the ui when http requests are made with one global spinner and/or cdk overlay? what happens when there are multiple http calls being made or if one fails etc?

we should aim to lazy load in data and update the ui as soon as it's available no?

Reply:
multiple it blocks until last http-call 
one component with cdk-overlay, mat-dialoge, custom overlay etc and mat spinner in the middle 
on every http error hide spinner, route error hide spinner. 
while lazy loading etc spinner shows as well.
I have a custom one readily available in another repo if you wanna see? Also examples on internet.

Reply:
im not so sure if its the best user experience to block the ui until last http-call resolves... might be worth getting some more thoughts from the trainee-ui devs... any thoughts @Judge40 @babul3012 @hihilary ?

Reply:
Not particularly keen on the idea of blocking the whole UI, we have that on part of the TIS app and it's not user friendly at all.